server/controllers/message.js

```diff
@@ -0,0 +1,164 @@
+const asyncHandler = require('express-async-handler');
+const decodeToken = require('../utils/decodeToken');
+
+const Message = require('../models/Message');
+const Conversation = require('../models/Conversation');
+
+// @route POST /conversation/create/:id
+// Create a conversation
+exports.newConversation = asyncHandler(async (req, res) => {
+  const recipient = req.params.id;
+  const { type, content } = req.body;
+  let decoded = decodeToken(req.cookies.token);
+  const userId = decoded.id;
+  try {
+    const existingConversation = await Conversation.find({
+      $and: [{ participants: { $in: userId } }, { participants: { $in: recipient } }, { deleted: false }],
+    });
+    if (existingConversation.length > 0) {
+      res.status(403).json({ error: 'Conversation already exists' });
+    } else {
+      const message = await Message.create({
+        sender: userId,
+        recipient,
+        sendDate: Date.now(),
+        content,
+        type,
+      });
+      await Conversation.create({
+        participants: [userId, recipient],
+        messages: [message],
+        mostRecentMsg: message,
+        unreadMsgs: 1,
+      });
+      res.status(200).json({
+        success: {
+          message: {
```

Comment:
We can send the newly created conversation as a response too, so that we can display the conversation on the FE

Reply:
I think that depends on where we start the conversation. As of now, we are planning on having a user initiate a conversation by first sending a message through a text field that will be in another user's profile details page. But I will make the suggested change just in case we add the ability for a user to start a conversation in the messages section.